tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/service/http/sample/SerializeComplexXmlTestCase.java

```diff
@@ -0,0 +1,100 @@
+/*
+ *  Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.ballerinalang.test.service.http.sample;
+
+import io.netty.handler.codec.http.HttpHeaderNames;
+import org.ballerinalang.test.service.http.HttpBaseTest;
+import org.ballerinalang.test.util.HttpClientRequest;
+import org.ballerinalang.test.util.HttpResponse;
+import org.ballerinalang.test.util.TestConstant;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+
+import static org.ballerinalang.jvm.XMLFactory.isEqual;
+import static org.ballerinalang.jvm.XMLFactory.parse;
+
+/**
+ * Test case for XML Serialization.
+ */
+@Test(groups = "http-test")
+public class SerializeComplexXmlTestCase extends HttpBaseTest {
+    private static final String XML_DATA = "<?xml version=\"1.0\"?>\n" +
```

Comment:
Would it be better to read from the `ComplexTestXmlSample.xml` itself for the assertion as well?
So we don't have to duplicate the same XML in two places